File: akka-diagnostics/src/test/scala/docs/StarvationDetectorDocSample.scala
/*
 * Copyright (C) 2025 Lightbend Inc. <https://akka.io>
 */

package docs

import akka.actor.typed.ActorSystem
import akka.actor.typed.scaladsl.Behaviors

object StarvationDetectorDocSample {

  val system: ActorSystem[_] = ActorSystem[Nothing](Behaviors.empty[Nothing], "Doc")

  // #other-dispatcher
  import akka.diagnostics.StarvationDetector

  StarvationDetector.checkDispatcher(system, dispatcherConfigPath = "my-dispatcher")
  // #other-dispatcher

}

File: akka-diagnostics/src/test/java/jdocs/StarvationDetectorDocSample.java
/*
 * Copyright (C) 2025 Lightbend Inc. <https://akka.io>
 */

package jdocs;

import akka.actor.typed.ActorSystem;
//#other-dispatcher
import akka.diagnostics.StarvationDetector;

//#other-dispatcher

public class StarvationDetectorDocSample {

  static void illustrateOtherDispatcher(ActorSystem<?> system) {
    //#other-dispatcher
    String dispatcherConfigPath = "my-dispatcher";
    StarvationDetector.checkDispatcher(system, dispatcherConfigPath);
    //#other-dispatcher
  }
}
